lib/hal/video.c

```diff
@@ -259,15 +262,32 @@ VIDEO_MODE XVideoGetMode(void)
 void XVideoInit(DWORD dwMode, int width, int height, int bpp)
 {
 	ULONG Step = 0;
-	DWORD dwFormat = (bpp == 32) ? VIDEO_BPP_32 : VIDEO_BPP_16;
-	int screenSize = width * height * (bpp/8);
+	DWORD dwFormat = 0;
+	int bytesPerPixel = (bpp+7)/8;
+	int pitch = width * bytesPerPixel;
+	int screenSize = pitch * height;
+
+	switch(bpp) {
+	case 15:
+		dwFormat = VIDEO_A1R5G5B5;
```

Comment:
bpp=15 was the sanest solution I could find to integrate this. This is also how it's handled in QEMU